artifacts/centosstream/centos-stream.go

```diff
@@ -91,7 +92,8 @@ func (c *centos) Inspect() (*api.ArtifactDetails, error) {
 
 	if checksum, exists := checksums[candidate]; exists {
 		return &api.ArtifactDetails{
-			SHA256Sum:            checksum,
+			Checksum:             checksum,
+			ChecksumHasher:       sha256.New(),
```

Comment:
```suggestion
			ChecksumHash:       sha256.New,
```

Can you do it like this?

Saves passing around the object, and is more idiomatic.

See e.g. https://pkg.go.dev/crypto/pbkdf2#Key